daemon/daemon_utils.go

```diff
@@ -52,3 +59,93 @@ func initialRuntime(baseDir string, runtimes map[string]types.Runtime) error {
 
 	return nil
 }
+
+// checkSnapshotter checks whether the given snapshotter is valid
+func checkSnapshotter(snapshotter string, containerdRootDir string) error {
+	var (
+		driverFound = false
+	)
+
+	fInfo, err := os.Stat(containerdRootDir)
+	if err != nil {
+		return fmt.Errorf("failed to stat containerd root dir: %v", err)
+	}
+
+	if !fInfo.IsDir() {
+		return fmt.Errorf("containerd root dir %s is not a directory", containerdRootDir)
+	}
+
+	infos, err := ioutil.ReadDir(containerdRootDir)
+	if err != nil {
+		return fmt.Errorf("failed to read dir %s: %v", containerdRootDir, err)
+	}
+
+	// read all snapshotter dir to make sure snapshotter is valid
+	for _, info := range infos {
+		found, err := checkSnapshotterDir(snapshotter, containerdRootDir, info)
+		if err != nil {
+			return err
+		}
+
+		if found {
+			driverFound = found
```

Comment:
I think it should break loop at here.

And no need to add more `found`, just `use driverFound`, like this:
```go
	for _, info := range infos {
                 var err error
		driverFound, err = checkSnapshotterDir(snapshotter, containerdRootDir, info)
		if err != nil {
			return err
		}

		if driverFound {
			return nil
		}
	}
```

Reply:
If a driver is found in containerd snapshotter, we also need to check other snapshotter dir whether it exists snapshots. If another snapshotter has snapshots, another snapshotter may be set last startup by pouchd. @rudyfly

Reply:
Not followed what you say, why should check other snapshotter

Reply:
For example, We start pouchd with overlayfs and then pull some images and create containers. Then we restart pouchd with brtfs. Overlayfs and btrfs are all supported by containerd, but the startup with btrfs should fail, otherwise previous images and containers may be in wrong use. It is important to make sure snapshotter to be consistent, except snapshots have been removed.  @Ace-Tang

Reply:
@wangforthinker Just to confirm, if we change snapshotter from `btrfs` to `overlayfs` in containerd, the containerd do what? refuse the behavior or just remove the containers or images created by previous snapshotter?